### ISSUE TITLE
Bottom dock(terminal) trigger cmd wrong height rendering (flickering)

### ISSUE THREAD

**Issue by d1y** [zed-industries/zed#8363]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

In fact, judging from the gif, when you open a terminal, it will flash highly.
![fix3](https://github.com/zed-industries/zed/assets/45585937/33afb814-41bc-4957-bc37-2931ad3e4776)

### Environment

Zed: v0.124.4 (Zed Preview)
OS: macOS 14.3.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by JosephTLyons** [comment]
I think the flickering is related to the same underlying issue as:

- https://github.com/zed-industries/zed/issues/8207

Ill leave this open and once a fix is in place for #8207, we can see if it clears this bug up.

**Comment by JosephTLyons** [comment]
I think this may be fixed in [v0.125.0-pre](https://zed.dev/releases/preview/0.125.0). Let's close this out and we can reopen it if it it comes back.

**Comment by Moshyfawn** [comment]
It may actually be a separate issue from #8207, and we may want to reopen it and close #8708 in its favour.

**Comment by d1y** [comment]
@JosephTLyons I believe this problem still exists

https://github.com/zed-industries/zed/assets/45585937/c3335631-1d35-47d6-8087-fc93ccab4b52

**Comment by d1y** [comment]
I also found that when I zoomed in to full screen, project-panel also had this problem

https://github.com/zed-industries/zed/assets/45585937/31230621-b591-41b1-a807-b88723f3d0d7

**Comment by d1y** [comment]
More video recording:

https://github.com/zed-industries/zed/assets/45585937/4e13d4e6-e945-4b27-bb36-3537b77f8e02

**Comment by Moshyfawn** [comment]
> More video recording:
> 
>  Screen.Recording.2024-03-06.at.00.25.57.mov

Not sure if the underlying issue is the same or not, but this one is tracked in #8598.

**Comment by Moshyfawn** [comment]
@d1y can you please confirm the version of Zed you're seeing this issue in?

**Comment by d1y** [comment]
> I can reproduce the issue on `0.125.2`.

Zed: v0.125.2 (Zed Preview)
OS: macOS 14.3.1
Memory: 16 GiB
Architecture: aarch64

**Comment by Congyuwang** [comment]
https://github.com/zed-industries/zed/assets/52687642/9c3822d6-e81e-4676-bc06-0b4d950322df


Zed: v0.124.8 (Zed)
OS: macOS 14.3.1
Memory: 16 GiB
Architecture: aarch64
